operatorcontrolplane/v1alpha1/types_conditioncheck.go

```diff
@@ -38,6 +39,26 @@ type PodNetworkConnectivityCheckSpec struct {
 	// +kubebuilder:validation:Pattern=`^\S+:\d*$`
 	// +required
 	TargetEndpoint string `json:"targetEndpoint"`
+
+	// TLS indicates if a tls handshake should be performed. Endpoints expecting
+	// a TLS handshake might log tls errors if a tls handshake is not attempted
+	// before closing the TCP connection. Defaults to true.
+	// +kubebuilder:default:true
+	// +optional
+	TLS bool `json:"tls,omitempty"`
```

Comment:
why not something like 

```go
type Spec struct{
  // optional, if specified, must contain tls.crt and tls.key.
  // if provided, the values will be used to make the tcp connection. This avoids log spam and allows additional status reports about credential viability.
  ClientCert SecretRef
}
```

Reply:
also, if the secret is missing, attempt the connection anyway.  indicate that in the godoc too.  this is good, but not fatal in cases where it is missing.

Reply:
Done.